### DISCUSSION TITLE
Add constant bitrate

### DISCUSSION THREAD

**Discussion by ghost** [turboderp/exllamav2#161] [Ideas]
There should be a constant bitrate variable option that would bypass the need for measuring, and that would allow for setting all layers to the same one desired size. I think it'd make sense for testing purposes and bypassing any dataset biases that might be there.

**Comment by ghost** [comment]
Another person also suggested a minimum bitrate setting too and I could see the benefits.

**Comment by turboderp** [comment]
My main worry with this is that it's going to create a flood of models with all different kinds of settings, defeating the purpose of having a single parameter that sets a target average bitrate and then letting the quantizer figure out where the budget is best spent.

Dataset bias is inevitable, because like GPTQ, EXL2 inherently relies on calibration. Setting a fixed bitrate not only negates one of the main strengths of the format but wouldn't actually remove the bias since the quantization would still be error-corrected, and the error is measured by doing inference on the calibration dataset.

If you want a completely unbiased format, that's just RTN quantization.

**Comment by ghost** [comment]
"since the quantization would still be error-corrected, and the error is measured by doing inference on the calibration dataset."
That explains behavior at higher quants.
I still think it'd be worth it for the peace of mind (especially since one of the main benefits of EXL2 is the inference speed, not just the variable bitrate), even if just in the form of the additional minimal bitrate. (To let it do its thing but also keep it from doing very low bitrates on certain layers.)
It's not like it's impossible to just change your measurements.json to whatever you want, but a bypass would be easier.
